### DISCUSSION TITLE
How to create a controllable accordion?

### DISCUSSION THREAD

**Discussion by hedleygois** [chakra-ui/chakra-ui#2619] [Q&A] [answered]
Hi guys,

I just started using Chakra-UI but I'm already struggling to create an ultra-simple controlled accordion since there's no `expanded/collapsed` prop to control it. I tried searching on `useControllable` but that didn't help much.

**Comment by chr-ge** [comment, marked as answer]
You can use the `index` prop to controll which one(s) is open.
```js
const [active, setActive] = useState([0])

return(
  <Accordion index={active}  allowMultiple>
    ....
  </Accordion>
)
```
You can also use [useControllableState](https://chakra-ui.com/docs/hooks/use-controllable#usecontrollablestate) if you prefer.

**Reply by hedleygois** [reply]
Thanks!

**Reply by subhamsaha-manu** [reply]
Hi @chr-ge,when I pass index array for the <Accordion> then clicking on the accordion button does not expand any particular accordion item. Here is a code sandbox link

https://codesandbox.io/s/spring-tdd-kbw4rc?file=/src/App.js

My actual use case is, I want to click a button on the accordion-item(which shows up on hover) that should fire a graphql mutation and at the end the particular accordion item should be expanded. In my production code I am handling the expansion using a state variable passing it to the index prop, just like your example. For simplicity I have just passed a hard coded value in the codesandbox example to show the issue.

**Reply by idoliki** [reply]
@subhamsaha-manu I know it's late reply but whoever needs it, you just need to add onChange listener to the <Accordion> component itself and setIndex to the prop onChange gives you.

**Reply by SaharshSamir** [reply]
This no longer works with v3.0.0
There is no export Accordion, it's AccordionRoot. And it doesn't accept an index prop
